crates/core/src/subscription/subscription.rs

```diff
@@ -233,200 +184,17 @@ fn evaluator_for_primary_updates(
                 };
                 let row_pk = pk_for_row(&row);
                 let row = row.data;
-                out.insert(row_pk, Op { op_type, row_pk, row });
+                out.push(Op { op_type, row_pk, row });
             }
         }
         Ok(out)
     }
 }
 
-impl QuerySet {
-    pub const fn new() -> Self {
-        Self(BTreeSet::new())
-    }
-
-    /// Queries all the [`StTableType::User`] tables *right now*
-    /// and turns them into [`QueryExpr`],
-    /// the moral equivalent of `SELECT * FROM table`.
-    pub(crate) fn get_all(relational_db: &RelationalDB, tx: &Tx, auth: &AuthCtx) -> Result<Self, DBError> {
-        let tables = relational_db.get_all_tables(tx)?;
-        let same_owner = auth.owner == auth.caller;
-        let exprs = tables
-            .iter()
-            .map(Deref::deref)
-            .filter(|t| t.table_type == StTableType::User && (same_owner || t.table_access == StAccess::Public))
-            .map(|src| SupportedQuery {
-                kind: query::Supported::Scan,
-                expr: QueryExpr::new(src),
-            })
-            .collect();
-
-        Ok(Self(exprs))
-    }
-
-    /// Incremental evaluation of `rows` that matched the [Query] (aka subscriptions)
-    ///
-    /// This is equivalent to run a `trigger` on `INSERT/UPDATE/DELETE`, run the [Query] and see if the `row` is matched.
-    ///
-    /// NOTE: The returned `rows` in [DatabaseUpdate] are **deduplicated** so if 2 queries match the same `row`, only one copy is returned.
-    #[tracing::instrument(skip_all)]
-    pub fn eval_incr(
```

Comment:
I have split the logic for eval and eval_incr depending on whether we are executing a single query or a batch of queries that return rows from the same table.